server/src/main/java/org/elasticsearch/common/util/concurrent/TaskExecutionTimeTrackingEsThreadPoolExecutor.java

```diff
@@ -153,20 +178,41 @@ public double pollUtilization() {
 
         lastTotalExecutionTime = currentTotalExecutionTimeNanos;
         lastPollTime = currentPollTimeNanos;
+
+        if (trackUtilizationEWMA) {
+            percentPoolUtilizationEWMA.addValue(utilizationSinceLastPoll);
+            // Test only tracking.
+            assert setUtilizationSinceLastPoll(utilizationSinceLastPoll);
+        }
```

Comment:
It feels odd to tie utilization polling to APM. I think they can be separate. For example, we can have a separate set of `lastTotalExecutionTime` and `lastPollTime` which should allow us compute the utlization in a different frequency than the APM?

Reply:
> If we're willing to wear the cost of a thread to do the polling

We don't really need a "new" thread for it, right? It seems the options so far is either polling from `AverageWriteLoadSampler` or`InternalClusterInfoService`, both have existing thread schedule.

Reply:
The AverageWriteLoadSampler is part of the auto-scaling reporting from data node to master, and runs every 1 second. The InternalClusterInfoService runs every 30 seconds (INTERNAL_CLUSTER_INFO_UPDATE_INTERVAL_SETTING).

An EWMA based on samples every 30 seconds would likely be unusable.

I started working on a Runnable that gets rescheduled on the GENERIC thread pool every 3 seconds, say.

> One thing to note though, if we poll more often than once a minute the utilisation > 100% issues will get worse.

We could cap the utilization at 100%?

Reply:
> We could cap the utilization at 100%?

I think that would just leave us with wildly inaccurate utilisation numbers. I think 60s is a reasonable interval for polling this stuff. I don't think the > 100% utilisation is really that significant when you look at the thread utilisation dashboards. It seems to render utilisation fairly accurately overall. We could experiment with 30s and see how that looks.